File: Ordination/Java/src/Tests/OrdinationTests.java
package Tests;

import static org.junit.Assert.*;

import java.util.Arrays;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import BubbleSort.BubbleSort;
import CoutingSort.CoutingSort;
import GnomeSort.GnomeSort;
import InsertionSort.InsertionSort;
import MergeSort.MergeSort;
import QuickSort.QuickSort;
import QuickSort.QuickSortMedianOfThree;
import SelectionSort.SelectionSort;
import sorting.AbstractSort;

public class OrdinationTests {
	public Integer[] evenSize = {3,5,6,8,9,7,4,2,1,0};
	public Integer[] oddSize = {3,9,7,8,5,4,6,2,1};
	public Integer[] empty = {};
	public Integer[] equals = {5,5,5,5,5,5};
	public Integer[] repeated = {6,8,7,6,5,4,2,6,5,8,6,9};
	public Integer[] ordered = {1,2,3,4,5,6,7,8,9};
	
	AbstractSort<Integer> implementation;
	@Before
	public void setUp(){
		this.implementation = new CoutingSort();
	}
	//Auxiliar Methods
	public void genericSort(Integer[] array){
		Integer[] copy = Arrays.copyOf(array, array.length);
		this.implementation.sort(array);
		Arrays.sort(copy);
		Assert.assertArrayEquals(array, copy);
	}
	public void invalidInput(Integer[] array,int leftIndex ,int rightIndex){
		Integer[] copy = Arrays.copyOf(array, array.length);
		this.implementation.sort(array,leftIndex,rightIndex);
		Assert.assertArrayEquals(array, copy);
	}
	@Test
	public void testevenSize(){
		genericSort(evenSize);
	}
	@Test
	public void tesoddSize(){
		genericSort(oddSize);
	}
	@Test
	public void testempty(){
		genericSort(empty);
	}
	@Test
	public void testEquals(){
		genericSort(equals);
	}
	@Test
	public void testRepetido(){
		genericSort(repeated);
	}
	@Test
	public void testOdernados(){
		genericSort(ordered);
	}
	
	@Test
	public void invalidLeftIndex(){
		invalidInput(evenSize,-1,evenSize.length);
		invalidInput(evenSize,evenSize.length +2,evenSize.length);
		
		invalidInput(oddSize,-1,oddSize.length);
		invalidInput(oddSize,oddSize.length +2,oddSize.length);
		
		invalidInput(empty,-1,empty.length);
		invalidInput(empty,empty.length +2,empty.length);
		
		invalidInput(repeated,-1,repeated.length);
		invalidInput(repeated,repeated.length +2,repeated.length);
		
		invalidInput(equals,-1,equals.length);
		invalidInput(equals,equals.length +2,equals.length);
		
		invalidInput(ordered,-1,ordered.length);
		invalidInput(ordered,ordered.length +2,ordered.length);
	}
	
	@Test
	public void invalidRightIndex(){
		invalidInput(evenSize,0,evenSize.length);
		invalidInput(evenSize,0,-5);
		
		invalidInput(oddSize,0,oddSize.length);
		invalidInput(oddSize,0,-5);
		
		invalidInput(empty,0,empty.length);
		invalidInput(empty,0,-5);
		
		invalidInput(repeated,0,repeated.length);
		invalidInput(repeated,0,-5);
		
		invalidInput(equals,0,equals.length);
		invalidInput(equals,0,-5);
		
		invalidInput(ordered,0,ordered.length);
		invalidInput(ordered,0,-5);
		
	}
}

File: Python/OrdinationTests.py
import unittest

from BubbleSort import BubbleSort
from SelectionSort import SelectionSort
from InsertionSort import InsertionSort
from GnomeSort import GnomeSort
from MergeSort import MergeSort
from QuickSort import QuickSort
from BogoSort import BogoSort


class OrdinationTest(unittest.TestCase):
	
	def setUp(self):
		self.oddSize = [9,8,7,6,5,4,3,2,1,0]
		self.evenSize = [9,8,7,6,5,4,3,2,1]
		self.empty = []
		self.equal = [5,5,5,5,5,5]
		self.repeated = [5,9,8,7,6,8,6,8,5,4,3,2]
		self.ordered = [1,2,3,4,5,6,7,8,9]
		
		
		#self.algorithm = BubbleSort()
		#self.algorithm = SelectionSort()
		#self.algorithm = InsertionSort()
		#self.algorithm = GnomeSort()
		#self.algorithm = MergeSort()
		#self.algorithm = QuickSort()
		self.algorithm = BogoSort()
	
	def genericTest(self,array):
		copy = array[:]
		self.algorithm.sort(array)
		copy.sort()
		self.assertEqual(copy,array)

	def invalidParameters(self,array,leftIndex,rightIndex):
		copy = array[:]
		self.algorithm.sort(array,leftIndex,rightIndex)
		self.assertEqual(copy,array)

	def testOrdinationOdd(self):
		self.genericTest(self.oddSize)
	def testOrdinationEven(self):
		self.genericTest(self.evenSize)
	def testOrdinationEmpty(self):
		self.genericTest(self.empty)
	def testOrdinationEqual(self):
		self.genericTest(self.equal)
	def testOrdinationRepeated(self):
		self.genericTest(self.repeated)
	def testOrdinationOrdered(self):
		self.genericTest(self.ordered)

	def testInvalidLeftIndex(self):
		self.invalidParameters(self.oddSize,-5,3)
		self.invalidParameters(self.evenSize,-5,3)
		self.invalidParameters(self.empty,-5,3)
		self.invalidParameters(self.equal,-5,3)
		self.invalidParameters(self.repeated,-5,3)
		self.invalidParameters(self.ordered,-5,3)
	
	def testInvalidRightIndex(self):
		self.invalidParameters(self.oddSize,0,len(self.oddSize))
		self.invalidParameters(self.evenSize,0,len(self.evenSize))
		self.invalidParameters(self.empty,0,len(self.empty))
		self.invalidParameters(self.equal,0,len(self.equal))
		self.invalidParameters(self.repeated,0,len(self.repeated))
		self.invalidParameters(self.ordered,0,len(self.ordered))

	def testInvalidLeftRight(self):
		self.invalidParameters(self.oddSize,5,5)
		self.invalidParameters(self.evenSize,5,3)
if __name__ == "__main__":
	unittest.main()
